2주차 리뷰

Account-4-Review opened the pull request:
## 🛠️ 과제 요약 
- 과제 요약 설명(ex 마이페이지 구현 완료)

## 📝 요구 사항과 구현 내용
- 구현 내용 1
- 구현 내용 2

## ✅ 피드백 반영사항
- 피드백 받은 내용과 반영된 결과 적기

## 💬 PR 포인트 & 궁금한 점
- 기능 관련 나누고 싶은 내용 or 리뷰 요구사항 작성

## 🔗 연관된 이슈
- 열린 이슈를 닫고싶으면 
- close #이슈번호 [이슈제목]
- 이슈를 닫지 않을 거라면 작성X

bung-dev committed:
:recycle: refactor : 수정 사항 리팩토링

bung-dev committed:
Merge pull request #171 from prgrms-web-devcourse-final-project/refactor/#110-refactor-chat

Refactor/#110 refactor chat

bung-dev committed:
:recycle: refactor : 수정 사항 리팩토링, 빌드그레들 추가

Si-rauis committed:
Merge pull request #172 from prgrms-web-devcourse-final-project/feat/#168-redis

:recycle: refactor : 수정 사항 리팩토링

bung-dev committed:
:bug: fix : 리뷰 수정 추가

usingjun committed:
:sparkles:  Feat : 채팅방 목록 조회 1차 수정

Si-rauis committed:
♻️ Refactor : 검색 필터링 인원수 비교 처리

bung-dev committed:
:sparkles: feat : 결제 취소 반환값 추가 및 예약 수정사항 적용

Si-rauis committed:
♻️ Refactor : ELB 건강 체크

Si-rauis committed:
♻️ Refactor : Reservation 응답값 추가 및 시큐리티 설정

Si-rauis committed:
Merge pull request #175 from prgrms-web-devcourse-final-project/feat/#168-redis

:sparkles: feat : 결제 취소 반환값 추가 및 예약 수정사항 적용

Account-4-Review commented on src/main/java/roomit/main/domain/business/controller/BusinessController.java:
```diff
@@ -13,38 +13,39 @@
-    @PutMapping("api/v1/business")
-    public void businessModify(@RequestBody @Valid BusinessUpdateRequest businessUpdateRequest
-                               ,@AuthenticationPrincipal CustomBusinessDetails customBusinessDetails){
+    @PutMapping("")
```
사소한거긴 한데, 사실 이런 경우엔 그냥 `@PutMapping` 을 써도 됩니다.

Account-4-Review commented on src/main/java/roomit/main/domain/business/repository/BusinessRepository.java:
```diff
@@ -1,15 +1,29 @@
+    @Query(value = "SELECT CASE WHEN COUNT(b) > 0 THEN true ELSE false END FROM Business b WHERE b.businessNum.value=:businessNum")
+    Boolean existsByBusinessNum(String businessNum);
+
+    @Query("""
```
존재 여부를 확인하는 건가요?
이 경우, `EXIST` 을 걸면 좋습니다.
한 개 검색이 되는 순간 쿼리를 종료시켜 버리거든요.

```java
@Query("""
	    SELECT CASE WHEN EXISTS (
	        SELECT 1 
	        FROM Business b
	        WHERE b.businessId = :senderId AND b.businessName = :senderName
	    ) THEN TRUE ELSE FALSE END
	""")
boolean existsByIdAndBusinessName(Long senderId, String senderName);
```

Account-4-Review commented on src/main/java/roomit/main/domain/business/repository/BusinessRepository.java:
```diff
@@ -1,15 +1,29 @@
+    Optional<Business> findByBusinessEmail(String businessEmail);
+
+    @Query(value = "SELECT CASE WHEN COUNT(b) > 0 THEN true ELSE false END FROM Business b WHERE b.businessName.value=:businessName")
+    boolean existsByBusinessName(String businessName);
```
이것만 primitive 네요.

Account-4-Review commented on src/main/java/roomit/main/domain/chat/chatmessage/dto/ChatMessageResponse.java:
```diff
@@ -0,0 +1,24 @@
+
+import java.time.LocalDateTime;
+
+@Builder
```
직접 Builder 를 쓸 일은 없겠죠?
(타 record 포함)

Account-4-Review commented on src/main/java/roomit/main/global/token/controller/LoginController.java:
```diff
@@ -0,0 +1,33 @@
+        return tokenService.memberLogin(loginRequest,response);
+    }
+
+    @PostMapping("/login/business")
```
ResponseStatus 추가

Account-4-Review commented on src/main/java/roomit/main/global/oauth2/dto/PROVIDER.java:
```diff
@@ -0,0 +1,7 @@
+package roomit.main.global.oauth2.dto;
+
+public enum PROVIDER {
```
네이밍 컨벤션 맞춰주세요.

Account-4-Review commented on src/main/java/roomit/main/global/oauth2/config/CustomSuccessHandler.java:
```diff
@@ -36,17 +45,19 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
 
-            String access = jwtUtil.createJwt("access", username, role, 60 * 60 * 60L);
-            String refresh = jwtUtil.createJwt("refresh", username, role, 24 * 60 * 60 * 60L);
+            String access = jwtUtil.createJwt("access", username, role, 1000 * 60 * 15L);
```
해당 값이 코드 전반에 계속 보이는데요.
상수만 담는 파일을 만들어서 해당 파일에 아예 상수로 정의하는게 좀 더 나을 것 같아요.

Account-4-Review commented on src/main/java/roomit/main/global/oauth2/config/CustomSuccessHandler.java:
```diff
@@ -36,17 +45,19 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
 
             CookieUtil.addCookie(response, "refresh", refresh, 60 * 60);
 
+            log.info(role);
```
?

Account-4-Review commented on src/main/java/roomit/main/global/oauth2/config/CustomSuccessHandler.java:
```diff
@@ -36,17 +45,19 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
+
             // Redirect URL
-            String redirectUrl = "http://localhost:3000/courses?accessToken=" + access; //프론트의 특정 페이지로 리다이렉션
+            String redirectUrl = oauthUrl + access +"&role="+role;
```
혹시 텍스트 조합하는 법에 대해 제가 설명을 안했을까요? (헷갈려서 ㅎㅎ;;)
혹시 그렇다면, 다음 멘토링 세션때 전달 드리겠습니다. 기억해 주세요!

Si-rauis replied:
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class BumblebeeStringUtil {
    public static String format(String format, Object... args) {
        return MessageFormatter.arrayFormat(format, args).getMessage();
    }
}
이 방식이 맞을까요?

Account-4-Review commented on src/main/java/roomit/main/global/config/security/SpringSecurityConfig.java:
```diff
@@ -89,14 +94,14 @@ public CorsConfiguration getCorsConfiguration(HttpServletRequest request) {
 
-                        configuration.setAllowedOrigins(Collections.singletonList("http://localhost:3000"));
+                        configuration.setAllowedOrigins(
+                            List.of("https://your-frontend-domain.com",corsUrl,  "http://127.0.0.1:5000"));
```
변수로 주입 필요

bung-dev committed:
:recycle: refactor : 결제 검증 예외 추가 , 예약 리팩토링

usingjun committed:
♻️ refactor : 채팅 ResponseStatus 추가

usingjun committed:
♻️ refactor : 채팅 redis 시간순 정렬에 따른 삭제 작업 실행 & 채팅 메시지 읽음 기능 추가

usingjun committed:
Merge branch 'develop' into refactor/#170-refactor-chat-redis

# Conflicts:
#	src/main/java/roomit/main/domain/chat/chatmessage/service/ChatService.java
#	src/main/java/roomit/main/domain/chat/chatroom/controller/ChatRoomController.java
#	src/main/java/roomit/main/domain/chat/chatroom/dto/ChatRoomBusinessResponse.java
#	src/main/java/roomit/main/domain/chat/chatroom/dto/ChatRoomMemberResponse.java
#	src/main/java/roomit/main/domain/chat/chatroom/dto/response/ChatRoomResponse.java
#	src/main/java/roomit/main/domain/chat/chatroom/entity/ChatRoom.java
#	src/main/java/roomit/main/domain/chat/chatroom/repositoroy/ChatRoomRepository.java
#	src/main/java/roomit/main/domain/chat/chatroom/service/ChatRoomService.java
#	src/test/java/roomit/main/domain/chat/service/ChatServiceIntegrationTest.java

Account-4-Review commented on src/main/java/roomit/main/global/service/FileLocationService.java:
```diff
@@ -0,0 +1,125 @@
+  private String bucketName;
+
+
+  public List<String> getImagesFromFolder(String folderPath) {
```
여기는 왜 indentation 이 2칸인가요?

Account-4-Review commented on src/main/java/roomit/main/global/service/FileLocationService.java:
```diff
@@ -0,0 +1,125 @@
+            // 해당 파일이 지정한 폴더에 속한 직접적인 파일인지 확인
+            if (key.startsWith(path + "/") && !key.substring(path.length() + 1).contains("/")) {
+              // 이미지 파일인지 확인 (확장자 필터)
+              if (key.endsWith(".jpg") || key.endsWith(".png") || key.endsWith(".jpeg")) {
```
이미지 확장자는 훨씬 다양하지 않나요?
상수로 빼고, Set으로 지정하는게 좋을 것 같아요.

Account-4-Review replied:
```java
  private String extractExtensionFromFileName(String fileName) {
    return fileName.substring(fileName.lastIndexOf(".") + 1);
  }
```

Account-4-Review commented on src/main/java/roomit/main/domain/chat/chatmessage/service/ChatService.java:
```diff
@@ -0,0 +1,152 @@
+        }
+
+        if (request.timestamp() == null) {
+            request = new ChatMessageRequest
```
request.withTimestamp(LocalDateTime.now()) 와 같이 만드는 건 어떨까요?

usingjun replied:
if (request.timestamp() == null) {
    request = request.withTimestamp(LocalDateTime.now());
}

이런식으로 만드는게 맞을까요?

Account-4-Review commented on src/main/java/roomit/main/domain/chat/chatmessage/service/ChatService.java:
```diff
@@ -0,0 +1,152 @@
+        ChatRoomDetailsDTO roomDetails = roomRepository.findRoomDetailsById(request.roomId())
+                .orElseThrow(ErrorCode.CHATROOM_NOT_FOUND::commonException);
+
+        boolean isSenderValid = (request.senderType().equals("business") && roomDetails.businessName().equals(request.sender()))
```
해당 파일의 로직은 전반적으로 리팩토링이 가능하네요. (멘토링 시간에 자세히 설명할게요)

Account-4-Review commented on src/main/java/roomit/main/domain/chat/chatmessage/service/ChatService.java:
```diff
@@ -0,0 +1,152 @@
+        ChatRoom room = roomRepository.findById(roomId)
+                .orElseThrow(ErrorCode.CHATROOM_NOT_FOUND::commonException);
+
+        System.out.println(senderType);
```
System.out.println 은 사용하지 말아주세요.

Account-4-Review commented on src/main/java/roomit/main/domain/chat/chatmessage/service/ChatMessageBatchProcessor.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Scheduled(fixedRate = 60000) // 1분마다 실행
+    public void flushMessages() {
+        List<Long> roomIds = chatRoomRepository.findAllRoomIds(); // Room ID 동적 조회
```
roomId 의 수가 끝도 없이 많을 수 있지 않을까요?

Account-4-Review commented on src/main/java/roomit/main/domain/chat/chatmessage/service/ChatService.java:
```diff
@@ -0,0 +1,152 @@
+        redisPublisher.publish(topic, request);
+
+        // Redis에 저장
+        saveMessageToRedis(request);
```
해당 로직은, "Redis에 1분간 발생한 모든 채팅의 내용이 전부 적재될 수 있다" 라는 전제하에 수행되는 로직으로 보여요.

그렇지만... 여러분들 서버의 Redis 메모리는 안녕하실까요?

usingjun replied:
ttl에 따라 일정 시간이 지나면 삭제되도록 했는데 그래도 문제가 될까요?

![image](https://github.com/user-attachments/assets/5ee045c9-de70-4a03-a4c7-16c0b461204d)

Account-4-Review commented on src/main/java/roomit/main/domain/chat/chatmessage/service/ChatService.java:
```diff
@@ -0,0 +1,152 @@
+
+            ChatMessage message = new ChatMessage(room, request);
+
+            messageRepository.save(message);
```
결국 이것도 모든 데이터에 대해 일일히 INSERT를 날리는거 아닌가요?
묶어서 처리한다고 해도, 결국 성능을 별 차이 없을 것으로 보여요.

usingjun replied:
인덱싱 처리하면 어떨까요?

Account-4-Review commented on src/main/java/roomit/main/domain/chat/chatroom/repositoroy/ChatRoomRepository.java:
```diff
@@ -0,0 +1,79 @@
+        END
+    ) 
+    FROM ChatRoom c
+    LEFT JOIN c.messages m
```
쿼리의 실행계획을 확인해 주세요. 쿼리 성능 병목으로 인해 발목잡힐 수 있을 것 같아보여요.

usingjun replied:
![image](https://github.com/user-attachments/assets/265dd1c6-0614-4c61-b471-4255b9640aea)
    
혹시  이렇게 쿼리를 바꿨는데 이렇게 하면 될까요?

Account-4-Review commented on src/main/java/roomit/main/domain/workplace/repository/WorkplaceRepository.java:
```diff
@@ -1,43 +1,60 @@
-            "AND w.workplace_longitude BETWEEN :left AND :right " +
-            "ORDER BY distance ASC", nativeQuery = true)
+    @Query(value = """
+        SELECT w.workplace_id, w.workplace_name, 
```
장소라면, 공간인덱스를 설정하고 관련 함수를 쓰는게 훨씬 성능이 좋고, 정확하지 않을까요?

usingjun replied:
공간인덱스 설정 후 관련함수 적용했습니다!

Account-4-Review reviewed (commented):
우선 러프하게 작성해봤습니다.
스타일이 너무 어긋난게 아니면 대부분 패스했고요, (시간 문제 때문에 ㅠㅠ) 러프하게 본지라 아직 로직을 구체적으로 살펴보진 않았습니다.
내일 조금 더 살펴보고 추가 리뷰 남길게요.

Si-rauis committed:
Merge pull request #174 from prgrms-web-devcourse-final-project/feat/#168-redis

feat : 결제 취소 반환값 추가 및 예약 수정사항 적용

Si-rauis committed:
Merge pull request #177 from prgrms-web-devcourse-final-project/refactor/#170-refactor-chat-redis

Refactor/#170 refactor chat redis

Si-rauis committed:
Merge pull request #179 from prgrms-web-devcourse-final-project/develop

develop 브랜치 clone

Song-min-geun committed:
:recycle: refactor : 수정사항

Song-min-geun committed:
:recycle: refactor : reservation refactor

Si-rauis committed:
♻️ Refactor : Business 쿼리문 수정

Commit:
✨ Feat : EmitterRepository 캐쉬 저장 추가

Commit:
✨ Feat : Notification 캐쉬 저장 추가

Commit:
♻️ Refactor : RecommendationService url 환경 변수 변경

# Conflicts:
#	src/main/java/roomit/main/domain/reservation/service/ReservationService.java

Commit:
♻️ Refactor : ReviewService에 있는 Notification 수정

Commit:
♻️ Refactor : ReviewServiceTest

Song-min-geun committed:
Update Reservation.java

Song-min-geun committed:
Update ReservationState.java

Si-rauis committed:
♻️ Refactor : 사진 확장자명 Set 상수 처리

CryingPerson committed:
Merge pull request #183 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

Song-min-geun committed:
Merge remote-tracking branch 'origin/develop' into develop

Commit:
♻️ Refactor : ReviewService 본인 예약 아닐시 수정

Commit:
✨ Feat : ReservationRepository findByIdWithMember 추가

Si-rauis committed:
Merge pull request #181 from prgrms-web-devcourse-final-project/refactor/#180-reservation-refactor

Refactor/#180 reservation refactor

Commit:
✨ Feat : ResponseNotificationDto 알림 Type 추가

Si-rauis committed:
♻️ Refactor : 스터디룸, 사업장 변수 길이 설정

Si-rauis committed:
♻️ Refactor : S3 파일 불러오는 오류 수정 및 Reissue 도메인 응답값 수정

Si-rauis committed:
Merge pull request #185 from prgrms-web-devcourse-final-project/feat/#168-redis-siyoung

Feat/#168 redis siyoung

Si-rauis committed:
🔨 Build : Docker 설정 및 하이브리드 필터링 서버 이미지화

Si-rauis committed:
🗑️ Chore : Ci/CD설정 변경

Song-min-geun committed:
Merge remote-tracking branch 'origin/develop' into develop

bung-dev committed:
:sparkles: feat : 분산락 추가

bung-dev committed:
Merge branch 'feat/#168-redis' of https://github.com/prgrms-web-devcourse-final-project/WEB1_2_BUMBLEBEE_BE into feat/#168-redis

usingjun committed:
♻️ refactor : 채팅 멘토링 공간좌표 적용(build.gradle 수정)

usingjun committed:
♻️ refactor : 채팅 멘토링 공간좌표 적용 util

usingjun committed:
♻️ refactor : 채팅 메시지 전송 시 읽음 상태 추가 & 채팅방 목록 조회에 마지막 채팅메시지, 읽음 여부 추가

usingjun committed:
♻️ refactor : 채팅 멘토링 공간좌표 적용

usingjun committed:
Merge branch 'develop' into refactor/#178-mentor

usingjun committed:
:bug: Bug : 사업장 테스트 주석처리

bung-dev committed:
:sparkles: feat : 예약 스케줄링 추가

bung-dev committed:
:sparkles: feat : 분산락,예약 스케줄링 추가

Si-rauis committed:
Merge pull request #184 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

usingjun committed:
Merge pull request #188 from prgrms-web-devcourse-final-project/refactor/#178-mentor

Refactor/#178 mentor 멘토링 적용

Song-min-geun committed:
Merge remote-tracking branch 'origin/develop' into develop

Song-min-geun committed:
:recycle: refactor : 인덱싱 추가, 페이징 추가

Song-min-geun committed:
:recycle: refactor : 어노테이션 삭제

Song-min-geun committed:
:recycle: refactor : redisConfig 추가

Si-rauis committed:
🗑️ Chore : CI/CD설정 변경

usingjun committed:
Merge pull request #189 from prgrms-web-devcourse-final-project/feat/#168-redis

✨ feat : 분산락,예약 스케줄링 추가

Si-rauis committed:
Merge branch 'develop' into feat/#168-redis-siyoung

Si-rauis committed:
Merge pull request #192 from prgrms-web-devcourse-final-project/feat/#168-redis-siyoung

Feat/#168 CI/CD 작업 및 하이브리드 필터링

usingjun committed:
:wrench: BugFix : 시간 직렬화 수정, 채팅 구독 성공

usingjun committed:
Merge pull request #193 from prgrms-web-devcourse-final-project/refactor/yongjun

:wrench: BugFix : 시간 직렬화 수정, 채팅 구독 성공

usingjun committed:
Merge pull request #194 from prgrms-web-devcourse-final-project/develop

🔧 BugFix : 시간 직렬화 수정, 채팅 구독 성공

Si-rauis committed:
Merge pull request #195 from prgrms-web-devcourse-final-project/develop

develop 브랜치 clone

Si-rauis committed:
♻️ Refactor : 사업장 DTO 수정

Si-rauis committed:
♻️ Refactor : 검색 필터 수정 및 예약 반환값 수정

Si-rauis committed:
♻️ Refactor : CI/CD 및 Docker설정

Commit:
♻️ Refactor : sub 변경

Si-rauis committed:
Merge pull request #196 from prgrms-web-devcourse-final-project/feat/#168-redis-siyoung

Deploy/  CI/CD및 Docker설정

Commit:
♻️ Refactor : Business 변경

Commit:
✨ Feat : NotificationService 추가

Commit:
✨ Feat : NotificationRepository 추가

Commit:
✨ Feat : PaymentsService 에 알림 기능 추가

Commit:
✨ Feat : ResponseNotificationReservationDto  추가

Commit:
♻️ Refactor : ReviewService 변경

Si-rauis committed:
♻️ Refactor : 검색 필터링 null 수정

Commit:
✨ Feat : NotificationController 추가

Commit:
✨ Feat : NotificationService 추가

Commit:
✨ Feat : PaymentsService 추가

Commit:
✨ Feat : ReservationNotificationRepository 및 ReservationNotification 추가

Commit:
♻️ Refactor : ResponseNotificationReservationDto 변경

Commit:
♻️ Refactor : SpringSecurityConfig 추가

Commit:
♻️ Refactor : sub 추가

Commit:
♻️ Refactor : ### 예약 알람조회 API 테스트 추가

Si-rauis committed:
Merge pull request #198 from prgrms-web-devcourse-final-project/develop

배포 2

Commit:
♻️ Refactor : ### 예약 알람조회 API 테스트 추가

Commit:
♻️ Refactor :Notification 수정

Commit:
♻️ Refactor :Notification 수정

Song-min-geun committed:
Merge branch 'develop' into refactor/#182-reservation-refactor

Si-rauis committed:
♻️ Refactor : 검색 필터링 좌표 거리 수정

지러적 좌표계 : 4326
투영 좌표계 : 5181

Commit:
♻️ Refactor :중복 코드 수정

Commit:
♻️ Refactor :getNotifications 서비스 수정

Commit:
♻️ Refactor :ReservationNotification 수정

Commit:
♻️ Refactor :ReservationNotification 수정

Commit:
♻️ Refactor :PaymentValidationResponse 수정

Commit:
♻️ Refactor :PaymentValidationResponse alrim 수정

Commit:
♻️ Refactor :PaymentValidationResponse alrim 수정

bung-dev committed:
:recycle: refactor : 멘토링 수정사항 리팩토링

Si-rauis committed:
🗑️ Chore : 사업장 메인페이지 로그

Si-rauis committed:
🗑️ Chore : Double 타입 수정

Si-rauis committed:
🗑️ Chore : 트리거 설정

Commit:
♻️ Refactor :PaymentValidationResponse alrim 수정

Si-rauis committed:
Merge pull request #202 from prgrms-web-devcourse-final-project/refactor/#199

:recycle: refactor : 멘토링 수정사항 리팩토링

Commit:
♻️ Refactor :ReviewService Test 주석 수정

Commit:
♻️ Refactor :Member Test 주석 수정

Commit:
♻️ Refactor : 소나큐브  주석 수정

Si-rauis committed:
Merge pull request #201 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

Si-rauis committed:
🗑️ Chore : sonarQube 주석

Si-rauis committed:
Merge branch 'develop' into feat/#168-redis-siyoung

Si-rauis committed:
Merge pull request #203 from prgrms-web-devcourse-final-project/feat/#168-redis-siyoung

Refactor/ 사업장 및 거리계산 좌표 수정

Si-rauis committed:
Merge pull request #205 from prgrms-web-devcourse-final-project/develop

3차 배포

Si-rauis committed:
♻️ Refactor : 사업장, 스터디룸 조회 필터링 수정

Si-rauis committed:
🗑️ Chore : Docker 파일 수정

Si-rauis committed:
🗑️ Chore : Oauth2 role 반환 수정

Si-rauis committed:
Merge pull request #206 from prgrms-web-devcourse-final-project/release

Refactor/ 자잘한 수정 사항 및 Docker파일 수정

usingjun committed:
♻️Refactor : 멘토링 코드 리팩토링

usingjun committed:
♻️Refactor : 채팅 메시지 시간 포맷팅

usingjun committed:
Merge branch 'develop' into refactor/207-refactor-채팅메시지-시간-형식-변환

Si-rauis committed:
✨ Feat : 하이브리드 필터링 기능 구현

Si-rauis committed:
Merge pull request #209 from prgrms-web-devcourse-final-project/refactor/207-refactor-채팅메시지-시간-형식-변환

Refactor/207 refactor 멘토링 리팩토링 & 채팅메시지 시간 형식 변환

Si-rauis committed:
Merge pull request #211 from prgrms-web-devcourse-final-project/refactor/#208-성능-테스트

✨ Feat : 하이브리드 필터링 기능 구현

Si-rauis committed:
Merge pull request #212 from prgrms-web-devcourse-final-project/develop

Deploy Version 4

usingjun committed:
♻️Refactor : workplace 수정

usingjun committed:
Merge branch 'develop' into fix/#210-fix-workplace-bugfix

usingjun committed:
:wrench: BugFix : workplace 생성 오류 수정

usingjun committed:
♻️Refactor : chatting 동적 subscribe

usingjun committed:
Merge pull request #213 from prgrms-web-devcourse-final-project/fix/#210-fix-workplace-bugfix

Fix/#210 fix workplace bugfix & chatting dynamic subscribe

usingjun committed:
Merge pull request #214 from prgrms-web-devcourse-final-project/develop

fix workplace bugfix & chatting dynamic subscribe

Commit:
♻️ Refactor : Member 연관관계 수정

# Conflicts:
#	src/main/java/roomit/main/domain/member/entity/Member.java

Commit:
✨ Feat : Member예약시 알림 기능 추가

Commit:
♻️ Refactor :Notification 알림 수정

Commit:
✨ Feat : 멤버 예약 알림 DTO 추가

Commit:
✨ Feat : 멤버 예약 알림 DTO 추가

Commit:
♻️ Refactor :리뷰 알림 DTO 수정

Commit:
♻️ Refactor :ReviewService 수정

Commit:
♻️ Refactor :PaymentsService 수정

Commit:
✨ Feat : 알림 조회 기능 추가

Commit:
🗑️ Chore : 테스트 코드 추가

Commit:
🗑️ Chore : 소나 테스트 통과하기 위한 커밋 추가

Commit:
🗑️ Chore : 소나 테스트 통과하기 위한 커밋 추가

Commit:
🗑️ Chore : 소나 테스트 통과하기 위한 커밋 추가

Commit:
🗑️ Chore : 소나 테스트 통과하기 위한 커밋 추가

Commit:
🗑️ Chore : NotificationService 에 캐쉬 저장 삭제

usingjun committed:
♻️Refactor : chatting 동적 subscribe

usingjun committed:
Merge branch 'develop' into fix/#210-fix-workplace-bugfix

usingjun committed:
Merge pull request #216 from prgrms-web-devcourse-final-project/fix/#210-fix-workplace-bugfix

Fix/#210 fix chat bugfix

usingjun committed:
Merge pull request #217 from prgrms-web-devcourse-final-project/develop

채팅 동적 매핑 오류 수정

Si-rauis committed:
Merge pull request #215 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

Si-rauis committed:
Merge pull request #218 from prgrms-web-devcourse-final-project/develop

Deploy Version 5

usingjun committed:
♻️Refactor : chatting senderType

usingjun committed:
Merge branch 'develop' into fix/#210-fix-workplace-bugfix

usingjun committed:
♻️Refactor : 사진 수정

Commit:
🗑️ Chore : NotificationDTO 수정

Commit:
🗑️ Chore : Notification 수정

Commit:
🗑️ Chore : Notification id 추가

Commit:
🗑️ Chore : 알림 위치 수정

Commit:
🗑️ Chore : 테스트 코드 수정

bung-dev committed:
Merge pull request #219 from prgrms-web-devcourse-final-project/fix/#210-fix-workplace-bugfix

Fix/#210 fix workplace bugfix

CryingPerson committed:
Merge pull request #220 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

bung-dev committed:
Merge pull request #221 from prgrms-web-devcourse-final-project/develop

Deploy Version 6

Commit:
✨ Feat : Review 알림 Entity 추가

Commit:
♻️ Refactor :Review value -> String 으로 수정

Commit:
♻️ Refactor :Review value -> String 으로 수정

Commit:
♻️ Refactor :Review value -> String 으로 수정

Commit:
♻️ Refactor :Test 수정

Commit:
🗑️ Chore : 테스트 코드 수정

Commit:
🗑️ Chore : 테스트 코드 수정

CryingPerson committed:
Merge pull request #223 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

CryingPerson committed:
Merge pull request #224 from prgrms-web-devcourse-final-project/develop

Deploy Version 7

Commit:
🗑️ Chore : 테스트 코드 수정

Commit:
🗑️ Chore : 테스트 코드 수정

CryingPerson committed:
Merge pull request #225 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

CryingPerson committed:
Merge pull request #226 from prgrms-web-devcourse-final-project/develop

Deploy Version 8

Song-min-geun committed:
Merge pull request #191 from prgrms-web-devcourse-final-project/refactor/#182-reservation-refactor

Refactor/#182 reservation refactor

Commit:
🗑️ Chore : 테스트 코드 수정

CryingPerson committed:
Merge pull request #227 from prgrms-web-devcourse-final-project/feature/#149-notification

🗑️ Chore : 테스트 코드 수정

CryingPerson committed:
Merge pull request #228 from prgrms-web-devcourse-final-project/develop

Deploy Version 9

usingjun committed:
:wrench: BugFix : 오류 수정

Commit:
🗑️ Chore : 테스트 코드 수정

CryingPerson committed:
Merge pull request #229 from prgrms-web-devcourse-final-project/feature/#149-notification

🗑️ Chore : 테스트 코드 수정

CryingPerson committed:
Merge pull request #230 from prgrms-web-devcourse-final-project/develop

Deploy Version 10

usingjun committed:
♻️ Refactor : 레디스에서 mysql로 저장되는 시간 단축 & 사업장 별로 다른 채팅방으로 로직 개선

usingjun committed:
♻️ Refactor : 사업자id로 사업장 조회 응답값 수정

usingjun committed:
Merge branch 'develop' into fix/#210-fix-workplace-bugfix

usingjun committed:
Merge pull request #231 from prgrms-web-devcourse-final-project/fix/#210-fix-workplace-bugfix

Refactor/ 채팅방 사업장별로 수정 & workplace 사업자id 조회 응답값 변경

usingjun committed:
Merge pull request #232 from prgrms-web-devcourse-final-project/develop

채팅방 및 사업장 수정

Commit:
🗑️ Chore : 기존 알림 서비스 고유 키값으로 변경후 나머지 String으로 변환

Commit:
🗑️ Chore : http 변환

CryingPerson committed:
Merge pull request #233 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

CryingPerson committed:
Merge pull request #234 from prgrms-web-devcourse-final-project/develop

Deploy Version 11

Si-rauis committed:
🗑️ Chore : SonarQube Workflow 제거

Si-rauis committed:
🗑️ Chore : 스프링 시큐리티 Header설정

usingjun committed:
♻️ Refactor : redis -> mysql로 빠르게 넘겨서 읽음처리 보완

usingjun committed:
Merge branch 'develop' into fix/#210-fix-workplace-bugfix

usingjun committed:
Merge pull request #235 from prgrms-web-devcourse-final-project/fix/#210-fix-workplace-bugfix

Refactor/#210 읽음 처리 보완

usingjun committed:
Merge pull request #236 from prgrms-web-devcourse-final-project/develop

채팅 보완 처리

usingjun committed:
♻️ Refactor : sonarcube 수정

usingjun committed:
Merge pull request #237 from prgrms-web-devcourse-final-project/fix/#210-fix-workplace-bugfix

sonarcube

usingjun committed:
:wrench: BugFix : 채팅기록 오류 수정

usingjun committed:
Merge branch 'develop' into fix/#210-fix-workplace-bugfix

usingjun committed:
Merge pull request #238 from prgrms-web-devcourse-final-project/fix/#210-fix-workplace-bugfix

Fix/#210 fix 채팅오류 수정

usingjun committed:
Merge pull request #240 from prgrms-web-devcourse-final-project/develop

release

Si-rauis committed:
♻️ Refactor : isRead 초기값 false

Commit:
♻️ Refactor :NotificationService에 FileLocationService 사진 추가

Commit:
♻️ Refactor :PaymentsService에 FileLocationService 사진 추가

CryingPerson committed:
Merge pull request #241 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

CryingPerson committed:
Merge pull request #242 from prgrms-web-devcourse-final-project/develop

Deploy Version 11

usingjun committed:
♻️ Refactor : 채팅 읽음 처리

usingjun committed:
Merge branch 'develop' into fix/#210-fix-workplace-bugfix

usingjun committed:
Merge pull request #243 from prgrms-web-devcourse-final-project/fix/#210-fix-workplace-bugfix

Refactor/#210 채팅 레디스 키 변경

Commit:
♻️ Refactor : Notification service 수정

Commit:
♻️ Refactor : Notification Entity 수정

Commit:
♻️ Refactor : Notification DTO 수정

CryingPerson committed:
Merge pull request #244 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

bung-dev committed:
:recycle: refactor : 결제 예약 상태 로직 수정

bung-dev committed:
Merge pull request #245 from prgrms-web-devcourse-final-project/refactor/#199

:recycle: Refactor : 결제 예약 상태 로직 수정

bung-dev committed:
Merge pull request #246 from prgrms-web-devcourse-final-project/develop

Deploy Version 11

Si-rauis committed:
♻️ Refactor : Review imageURL 수정

bung-dev committed:
Deploy Version 12버전 Develop 브랜치로 Clone

Deploy Version 12버전 Develop 브랜치로 Clone

bung-dev committed:
:recycle: refactor : 리뷰 url 수정

bung-dev committed:
:recycle: Refactor : 리뷰 url 수정

:recycle: Refactor : 리뷰 url 수정

Si-rauis committed:
♻️ Refactor : FileLocationService 의존 추가

bung-dev committed:
Merge pull request #249 from prgrms-web-devcourse-final-project/refactor/#199

Deploy Version 13

Si-rauis committed:
♻️ Refactor : 리뷰 내용 최대 글자 100

Si-rauis committed:
♻️ Refactor : Workplace 거리별 조회 최대 10개 조회

Si-rauis committed:
Merge branch 'refs/heads/develop' into refactor/#199

Si-rauis committed:
Merge pull request #250 from prgrms-web-devcourse-final-project/release

Deploy Version 13버전 develop 브랜치로 clone

Si-rauis committed:
Merge pull request #251 from prgrms-web-devcourse-final-project/refactor/#199

Refactor/ 거리별 조회 최대 10개 , 리뷰 글자수 100

Commit:
🐛 Bug : Notificationservice 오류수정

Commit:
🐛 Bug : PaymentsService 오류수정

Commit:
🐛 Bug : ReviewService 오류수정

CryingPerson committed:
Merge pull request #252 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

CryingPerson committed:
Merge pull request #253 from prgrms-web-devcourse-final-project/develop

Deploy Version 14

Commit:
✨ Feat : 사업자 회원 성능 테스트 기능 확인

Commit:
♻️ Refactor : ReviewService 수정

Commit:
♻️ Refactor : ResponseNotificationDto 생성 날짜 수정

Commit:
♻️ Refactor : ResponseNotificationReservationDto 생성 날짜 수정

Commit:
♻️ Refactor : ResponseNotificationReservationMemberDto 생성 날짜 수정

CryingPerson committed:
Merge pull request #254 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

CryingPerson committed:
Merge pull request #255 from prgrms-web-devcourse-final-project/develop

Deploy Version 15

Commit:
♻️ Refactor : AI 모델 최신 동기화

Commit:
♻️ Refactor : k6 테스트

CryingPerson committed:
Merge pull request #256 from prgrms-web-devcourse-final-project/feature/#149-notification

Feature/#149 notification

bung-dev committed:
Merge pull request #257 from prgrms-web-devcourse-final-project/develop

Deploy Version 16

bung-dev committed:
:recycle: refactor : 특정 사업자 예약 확인 응답값 수정

bung-dev committed:
Merge branch 'refactor/#199' of https://github.com/prgrms-web-devcourse-final-project/WEB1_2_BUMBLEBEE_BE into refactor/#199

bung-dev committed:
Merge pull request #258 from prgrms-web-devcourse-final-project/refactor/#199

♻️ Refactor : 특정 사업자 예약 확인 응답값 수정

bung-dev committed:
Deploy Version 18

Deploy Version 18

Si-rauis committed:
Merge pull request #260 from prgrms-web-devcourse-final-project/release

Release에서 develop로 PR
